src/ModelContextProtocol/Diagnostics.cs

```diff
@@ -89,14 +86,17 @@ private static void InjectContext(object? message, string key, string value)
             case JsonRpcNotification notification:
                 parameters = notification.Params;
                 break;
-
-            default:
```

Comment:
I *think* this case covered response, so we should probably keep it?

Reply:
`default: break;` is a nop

Reply:
Though this section would also arguably be simpler as a switch expression:
```C#
JsonNode? parameters = message switch
{
    JsonRpcRequest request => request.Params,
    JsonRpcNotification notification => notification.Params,
    _ => null,
};
```